modules/crypto/src/lib/crypto/crypto-hash-transform.js

```diff
@@ -0,0 +1,44 @@
+import * as CryptoJS from 'crypto-js';
+
+export default class CryptoHashTransform {
+  static async hash(input, options) {
```

Comment:
The idea is that all hashing Transforms are guaranteed to always provide at least one method. Since some transforms can only do async hashing, the async method is the only one we can always provide. 

For Transforms that can do synchronous hashing, this just becomes a promise wrapper for `hashSync`, as you pointed out.